jubilant/_juju.py

```diff
@@ -11,6 +11,7 @@
 import tempfile
 import time
 from collections.abc import Callable, Iterable, Mapping
+from contextlib import nullcontext
```

Comment:
Our [style](https://github.com/canonical/operator/blob/main/STYLE.md#import-modules-not-objects) (apart from typing, which includes collections.abc) is to `import module` rather than `from module import thing`.
```suggestion
import contextlib
```